src/contracts/core/AllocationManager.sol

```diff
@@ -269,28 +308,75 @@ contract AllocationManager is
         // if the effect timestamp is not reached, return false
         if (block.timestamp < mInfo.effectTimestamp) {
             return (false, mInfo);
-        } else if (mInfo.effectTimestamp == 0) {
-            // if the effect timestamp is 0, there is no pending modification, so we can avoid the below logic
-            return (true, mInfo);
         }
 
         // update stored magnitudes
-        if (mInfo.pendingMagnitudeDelta > 0) {
-            mInfo.currentMagnitude += uint64(uint128(mInfo.pendingMagnitudeDelta));
-        } else {
-            uint64 magnitudeDelta = uint64(uint128(-mInfo.pendingMagnitudeDelta));
-            mInfo.currentMagnitude -= magnitudeDelta;
-            // if this is a deallocation, reduce the encumbered magnitude
-            encumberedMagnitude[operator][strategy] -= magnitudeDelta;
+        mInfo.currentMagnitude = _addInt128(mInfo.currentMagnitude, mInfo.pendingMagnitudeDelta);
+        mInfo.effectTimestamp = 0;
+        if (mInfo.pendingMagnitudeDelta < 0) {
+            // if this was a deallocation, update the encumbered magnitude
+            _updateEncumberedMagnitude({
+                operator: operator,
+                strategy: strategy,
+                encumberedMagnitudeDelta: mInfo.pendingMagnitudeDelta
+            });
         }
         mInfo.pendingMagnitudeDelta = 0;
 
         // update the magnitude info in storage
-        _operatorMagnitudeInfo[operator][strategy][operatorSetKey] = mInfo;
+        _setMagnitudeInfo({
+            operator: operator,
+            strategy: strategy,
+            operatorSetKey: operatorSetKey,
+            mInfo: mInfo,
+            isCompletion: true
+        });
 
         return (true, mInfo);
     }
 
+    function _setMagnitudeInfo(
+        address operator,
+        IStrategy strategy,
+        bytes32 operatorSetKey,
+        MagnitudeInfo memory mInfo,
+        bool isCompletion
+    ) internal {
+        MagnitudeInfo memory mInfoStored = _operatorMagnitudeInfo[operator][strategy][operatorSetKey];
+        OperatorSet memory operatorSet = _decodeOperatorSet(operatorSetKey);
+
+        // if the current magnitude has been updated, this is a slash, emit an event
+        if (mInfoStored.currentMagnitude != mInfo.currentMagnitude && !isCompletion) {
```

Comment:
pendingMagnitudeDelta is also changed in a slash to pending deallocation? 

Should be something like:
``` solidity
// Completions have already had magnitude info emitted from its initial deallocation
if (!isCompletion) {
       if (mInfoStored.currentMagnitude != mInfo.currentMagnitude) ....
       else if (mInfoStored.pendingMagnitudeDelta != mInfo.pendingMagnitudeDelta) ...
}
```

Reply:
both current and pending are changed, will wrap in `if (!isCompletion)` tho

Reply:
Ah I read this wrong since you’re also updating the mag of the pending which is still completable at the effect timestamp. This is correct

Reply:
Wrapping conditional cleaner tho, ince